### PR TITLE
ORLP-173 Controllers with url deck for admin fixed

### PR DESCRIPTION
**Pull request by Tamara20**
In controller method I fixed mapping now it is the save url, and in method create, and fix mistake with deck and newDeck, build link for this now. I separate some methods temporarily for example createNewDeckAdmin and createNewDeck in server, because it is used in other method of controller which is not mine.

### CODE REVIEW

File: src/main/java/com/softserve/academy/spaced/repetition/controller/DeckController.java [hunk 1]
```diff
@@ -120,7 +120,8 @@
     @Auditable(action = AuditingAction.CREATE_DECK_IN_COURSE)
     @PostMapping(value = "/api/category/{category_id}/courses/{course_id}/decks")
     @PreAuthorize(value = "@accessToUrlService.hasAccessToCourse(#category_id, #course_id)")
-    public ResponseEntity<DeckPublicDTO> addDeckToCourse(@RequestBody Deck deck, @PathVariable Long category_id, @PathVariable Long course_id) {
+    public ResponseEntity<DeckPublicDTO> addDeckToCourse(@RequestBody Deck deck,
+                                                         @PathVariable Long category_id, @PathVariable Long course_id) {
```
Comment (b0hdan) [inline src/main/java/com/softserve/academy/spaced/repetition/controller/DeckController.java]: Rename the variables in the camelCase, please.

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java [hunk 2]
```diff
@@ -104,8 +112,21 @@ public void createNewDeck(Deck newDeck, Long category_id) throws NotAuthorisedUs
...
         deck.setCategory(categoryRepository.findById(category_id));
         deck.setDeckOwner(user);
-        Deck save = deckRepository.save(deck);
-        newDeck.setId(save.getId());
+        Deck savedDeck = deckRepository.save(deck);
+        newDeck.setId(savedDeck.getId());
+    }
+
+    @Transactional
+    public Deck createNewDeckAdmin(Deck newDeck) throws NotAuthorisedUserException {
```
Comment (b0hdan) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: You have to add this deck to admin's cabinet also, haven't you?
And to delete it from there as well then.

File: src/main/java/com/softserve/academy/spaced/repetition/controller/DeckController.java (hunk 1 above)
Comment (Tamara20) [reply src/main/java/com/softserve/academy/spaced/repetition/controller/DeckController.java]: It is not my method

Comment (b0hdan) [reply src/main/java/com/softserve/academy/spaced/repetition/controller/DeckController.java]: Well, okay.

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java (hunk 2 above)
Comment (Tamara20) [reply src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: yes it is true. admin is a user.

Comment (Tamara20) [reply src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: We need to decide what to do with the folder and after that I will correct it

Comment (b0hdan) [reply src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: What's wrong with the folder?

Comment (Tamara20) [reply src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: The big loop - get all forders users

Comment (b0hdan) [reply src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: Sorry, I still cannot understand you. Why can't you just add them to the only user's folder?

Comment (Tamara20) [reply src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: When I will delete from folder I will use mathod with big loop. Andriy told about it. So I'll use it. ok.

File: src/main/java/com/softserve/academy/spaced/repetition/controller/DeckController.java [hunk 3]
```diff
@@ -178,7 +183,8 @@ public ResponseEntity updateDeckForAdmin(@RequestBody Deck deck, @PathVariable L
 
     @Auditable(action = AuditingAction.DELETE_DECK_ADMIN)
     @DeleteMapping(value = "/api/admin/decks/{deckId}")
-    public ResponseEntity deleteDeckForAdmin(@PathVariable Long deckId) {
+    public ResponseEntity deleteDeckForAdmin(@PathVariable Long deckId)throws NotAuthorisedUserException {
```
Comment (b0hdan) [inline src/main/java/com/softserve/academy/spaced/repetition/controller/DeckController.java]: Shouldn't this method be transactional? I am not sure but I think if the deletion of the deck generally wasn't successful, the deletion from the folder should be cancelled as well.

Comment (Tamara20) [reply src/main/java/com/softserve/academy/spaced/repetition/controller/DeckController.java]: ok

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java (hunk 2 above)
Comment (b0hdan) [reply src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: I've understood but I don't know what to do. In any case we're likely to forget to implement it later.

File: src/main/java/com/softserve/academy/spaced/repetition/controller/DeckController.java (hunk 3 above)
Comment (antonbakalets) [reply src/main/java/com/softserve/academy/spaced/repetition/controller/DeckController.java]: Transactional should be on service level.

**Pull request merged**